### ISSUE TITLE
tslint complaining about semicolon for new line in regexp 

### ISSUE THREAD

**Issue by Maxwell2022** [palantir/tslint#3013]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function nl2br(str:string, is_xhtml:boolean) {
  var breakTag = is_xhtml || typeof is_xhtml === 'undefined' ? '<br />' : '<br>';
  var re = new RegExp('([^>\r\n]?)(\r\n|\n\r|\r|\n)', 'g');
  return (str + '').replace(re, '$1' + breakTag + '$2');
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-duplicate-variable": true,
    "no-unused-variable": [
      true
    ],
    "semicolon": [true, "always"]
  },
  "rulesDirectory": [
    "node_modules/tslint-eslint-rules/dist/rules"
  ]
}
```

#### Actual behavior

```
[16:33:28]  tslint: src/libraries/utils.ts, line: 7
            Missing semicolon

       L6:  |
       L7:  |\n)', 'g');
       L8:    return (str + '').replace(re, '$1' + breakTag + '$2');
```

#### Expected behavior

tslint should not complain about semicolon

**Comment by IllusionMH** [comment]
Can you specify on what OS you can reproduce this problem and what line endings does this file has.
For some reason it looks like `\r` are converted to actual line breaks.

**Comment by Maxwell2022** [comment]
Sorry for the late reply, I'm using OSX Sierra, the line break is `\n`

The only way I found to display line break is to use `od` command:
```
od -c src/libraries/utils.ts
0000000    /   *       h   t   t   p   s   :   /   /   g   i   t   h   u
0000020    b   .   c   o   m   /   p   a   l   a   n   t   i   r   /   t
0000040    s   l   i   n   t   /   i   s   s   u   e   s   /   3   0   1
0000060    3       *   /  \n  \n   f   u   n   c   t   i   o   n       n
0000100    l   2   b   r   (   s   t   r   :   s   t   r   i   n   g   ,
0000120        i   s   _   x   h   t   m   l   :   b   o   o   l   e   a
0000140    n   )       {  \n           v   a   r       b   r   e   a   k
0000160    T   a   g       =       i   s   _   x   h   t   m   l       |
0000200    |       t   y   p   e   o   f       i   s   _   x   h   t   m
0000220    l       =   =   =       '   u   n   d   e   f   i   n   e   d
0000240    '       ?       '   <   b   r       /   >   '       :       '
0000260    <   b   r   >   '   ;  \n           v   a   r       r   e
0000300    =       n   e   w       R   e   g   E   x   p   (   '   (   [
0000320    ^   >   \   r   \   n   ]   ?   )   (   \   r   \   n   |   \
0000340    n   \   r   |   \   r   |   \   n   )   '   ,       '   g   '
0000360    )   ;  \n           r   e   t   u   r   n       (   s   t   r
0000400        +       '   '   )   .   r   e   p   l   a   c   e   (   r
0000420    e   ,       '   $   1   '       +       b   r   e   a   k   T
0000440    a   g       +       '   $   2   '   )   ;  \n   }  \n  \n   e
0000460    x   p   o   r   t       d   e   f   a   u   l   t       {
0000500    n   l   2   b   r   :       n   l   2   b   r       }  \n
0000517
```

**Comment by IllusionMH** [comment]
Wasn't able to reproduce this on Windows. Will take a look when will be able to borrow Mac

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
